### PR TITLE
Updated Events section

### PR DESCRIPTION
**Pull request by Aeltumn**
I'm going to give every section a pass getting it up-to-date with 1.14 whenever I've got some time.

Here are some tiny improvements to the Events section, I've added some information and an example on how to use the properties of the Mod.EventBusSubscriber annotation to your advantage to register client-only events. This will hopefully aid people separating their client/common classes because they use the annotation instead.

### CODE REVIEW

File: docs/events/intro.md [hunk 1]
```diff
@@ -43,8 +43,21 @@ which must be registered like this: `MinecraftForge.EVENT_BUS.register(MyStaticF
...
+	@SubscribeEvent
+	public static void drawLast(RenderWorldLastEvent event) {
+		System.out.println("Drawing!");
+	}
+}
+```
+
 !!! note
-    This does not register an instance of the class; it registers the class itself (i.e. the event handling methods must be static).
+    This does not register an instance of the class; it registers the class itself (i.e. the event handling methods must be static.
```
Comment (tterrag1098) [inline docs/events/intro.md]: Don't remove that paren

**Review by tterrag1098 — CHANGES_REQUESTED** [review]
This could use a bit more documentation on the mod bus, and which types of events use it.

File: docs/events/intro.md [hunk 2]
```diff
@@ -84,3 +86,17 @@ Sub Events
 ----------
 
 Many events have different variations of themselves, these can be different but all based around one common factor (e.g. `PlayerEvent`) or can be an event that has multiple phases (e.g. `PotionBrewEvent`). Take note that if you listen to the parent event class, you will receive calls to your method for *all* subclasses.
+
+Mod Event Bus
+-------------
+
+The mod event bus is primarily meant for listening to lifecycle events in which mods should initialize. Many of these events are also ran in parallel so mods can be initialized at the same time. This does mean you can't directly execute code from other mods in these events, use the `InterModEnqueueEvent` and `InterModProcessEvent` events for that.
```
Comment (tterrag1098) [inline docs/events/intro.md]: The mod event bus is used for all events that are only called during the load process. This includes lifecycle events yes, but also registry events, and other misc load-time events like color handlers and model bake etc.

Comment (tterrag1098) [reply docs/events/intro.md]: Also perhaps mention DeferredWorkQueue

**Review by tterrag1098 — CHANGES_REQUESTED** [review]

**Comment by Aeltumn** [comment]
I'm not super familiar with the mod event bus, didn't have a clue which events where fired on which bus before I looked into it today, so the information might be a bit lacklustre.

**Review by tterrag1098 — APPROVED** [review]

**Pull request merged**